### PR TITLE
Feature: add guide to dungeon hub races

### PR DESCRIPTION
**Pull request by catgirlseraid**
## Dependencies
- https://github.com/hannibal002/SkyHanni-REPO/pull/153

## What
Add guide to dungeon hub races for all three races. Only works for Nothing; No return races

<details>
<summary>Images</summary>

image

</details>

## Changelog New Features
+ Added dungeon hub race waypoints. - seraid
    * Only works for Nothing; No return races.

## Changelog Technical Details
+ Added NamedParkourJson. - seraid

### CODE REVIEW

**Review by CalMWolfs — APPROVED** [review]

File: src/main/java/at/hannibal2/skyhanni/data/jsonobjects/repo/NamedParkourJson.kt
Comment (CalMWolfs) [inline src/main/java/at/hannibal2/skyhanni/data/jsonobjects/repo/NamedParkourJson.kt:1]: This should not be a map of maps, NamedParkourJson should only be `Map<String, ParkourJson>` and then a further `DungeonHubRacesJson` data class should hold a `Map<String, NamedParkourJson>`

Another option is to scrap the idea of making a NamedParkourJson and just making a DungeonHubRacesJson and adding the relevant fields into here, could maybe make an enum for race type and use those as some of the keys

**Review by CalMWolfs — CHANGES_REQUESTED** [review]

**Review by CalMWolfs — APPROVED** [review]

File: src/main/java/at/hannibal2/skyhanni/features/dungeon/DungeonsRaceGuide.kt [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+            it.value.lookAhead = config.lookAhead.get() + 1
+        }
+    }
+
+    @SubscribeEvent
+    fun onRenderWorld(event: LorenzRenderWorldEvent) {
+        if (!isEnabled()) return
+        if (currentRace == null) return
+
+        parkourHelpers[currentRace]?.render(event)
```
Comment (hannibal002) [inline src/main/java/at/hannibal2/skyhanni/features/dungeon/DungeonsRaceGuide.kt:87]: can we store the ParkourHelper instance in the variable currentRace? that way we dont need to do one map lookup each frame

**Review by hannibal002 — DISMISSED** [review]
works fine, except it breaks on every ability use (e.g. aote teleport). this is part of ParkourHelper, maybe you can fix it?

**Comment by CalMWolfs** [comment]
> works fine, except it breaks on every ability use (e.g. aote teleport). this is part of ParkourHelper, maybe you can fix it?

In my opinion this is not a problem as the feature is not made to work for the later races. The whole feature is written to only work with Nothing at all; No return races. In these races you cannot use abilities so this will never break. This is made as just a little guide for beginners as once you are at the point of using abilities and pearls these waypoints do not provide much use.

**Review by hannibal002 — APPROVED** [review]

**Pull request merged**